File: apps/backend-django/usuarios/dtos/sesion.py
"""Datos que devuelve el inicio de sesión."""

from dataclasses import dataclass

from usuarios.models import Usuario


@dataclass(frozen=True)
class SesionDTO:
    """Una sesión recién abierta: quién entró y con qué tokens."""

    usuario: Usuario
    token_de_acceso: str
    token_de_refresco: str

File: apps/frontend-react/src/modulos/autenticacion/dtos/sesion.js
import { usuarioDesdeApi } from './usuario'

/** Lo que manda el formulario de acceso. El correo viaja sin espacios; la contraseña, tal cual. */
export const credencialesHaciaApi = ({ correo, contrasena }) => ({
  correo: correo.trim(),
  contrasena,
})

/** Lo que manda «Salir»: el refresco que el servidor tiene que anular. */
export const cierreHaciaApi = (tokenDeRefresco) => ({ refresco: tokenDeRefresco })

/** Una sesión recién abierta: los dos tokens y quién entró. */
export const sesionDesdeApi = (sesion) => ({
  tokenDeAcceso: sesion.acceso,
  tokenDeRefresco: sesion.refresco,
  usuario: usuarioDesdeApi(sesion.usuario),
})
